### PR TITLE
Add recursive join node filter

### PR DESCRIPTION
**Pull request by andyfengHKU**

### CODE REVIEW

File: src/processor/operator/recursive_extend/recursive_join.cpp [hunk 1]
```diff
@@ -189,6 +190,7 @@ void RecursiveJoin::computeBFS(ExecutionContext* context) {
         } else {
             // Otherwise move to the next frontier.
             bfsState->finalizeCurrentLevel();
+            scanFrontier->setNodePredicateExecFlag(bfsState->isLastLevel());
```
Comment (andyfengHKU) [inline src/processor/operator/recursive_extend/recursive_join.cpp:193]: This is not needed. Simply set Flag to False.

File: src/binder/bind/bind_graph_pattern.cpp [hunk 2]
```diff
@@ -360,19 +361,43 @@ std::shared_ptr<RelExpression> Binder::createRecursiveQueryRel(const parser::Rel
...
+    if (recursivePatternInfo->whereExpression != nullptr) {
+        auto recursivePatternPredicate =
+            expressionBinder.bindExpression(*recursivePatternInfo->whereExpression);
+        for (auto& predicate : recursivePatternPredicate->splitOnAND()) {
+            auto expressionCollector = ExpressionCollector();
+            auto dependentVariableNames = expressionCollector.getDependentVariableNames(predicate);
+            auto dependOnNode = dependentVariableNames.contains(node->getUniqueName());
+            auto dependOnRel = dependentVariableNames.contains(rel->getUniqueName());
+            if (dependOnNode && dependOnRel) {
+                throw BinderException(stringFormat("Cannot evaluate {} in recursive pattern {}.",
```
Comment (ray6080) [inline src/binder/bind/bind_graph_pattern.cpp:385]: Escape LCOV check here.

File: src/planner/plan/append_extend.cpp [hunk 3]
```diff
@@ -140,8 +141,10 @@ void QueryPlanner::appendRecursiveExtend(std::shared_ptr<NodeExpression> boundNo
     appendAccumulate(AccumulateType::REGULAR, plan);
     // Create recursive plan
     auto recursivePlan = std::make_unique<LogicalPlan>();
-    createRecursivePlan(recursiveInfo->node, recursiveInfo->nodeCopy, recursiveInfo->rel, direction,
-        recursiveInfo->predicates, *recursivePlan);
+    //    createRecursivePlan(recursiveInfo->node, recursiveInfo->nodeCopy, recursiveInfo->rel,
```
Comment (ray6080) [inline src/planner/plan/append_extend.cpp:144]: Remove commented out lines.

File: src/processor/operator/recursive_extend/recursive_join.cpp (hunk 1 above)
Comment (ray6080) [reply src/processor/operator/recursive_extend/recursive_join.cpp:193]: Also remove `isLastLevel()`.

**Review by ray6080 — APPROVED** [review]

**Pull request merged**